tests/sdk/nodejs/nodejs_test.go

```diff
@@ -1802,3 +1802,104 @@ func simluateClusterKubeconfig(t *testing.T, numOfClusters int) ([]string, error
+			var ok bool
+			depName, ok = stackInfo.Outputs["deploymentName"].(string)
+			assert.True(t, ok)
+			depNS, ok = stackInfo.Outputs["deploymentNamespace"].(string)
+			assert.True(t, ok)
+
+			// Validate we applied the deployment with the correct name and namespace and spec.
+			dep, err := tests.Kubectl("get deployment -o yaml -n", depNS, depName)
+			assert.NoError(t, err)
+			assert.NotContains(t, string(dep), "Error from server (NotFound)")
+
+			// Validate image of deployment.
+			depImage, err := tests.Kubectl("get deployment -o=jsonpath='{.spec.template.spec.containers[0].image}' -n", depNS, depName)
+			assert.NoError(t, err)
+			assert.Equal(t, "'nginx:1.25.2'", string(depImage))
+
+			// Validate deployment replicas.
+			depReplicas, err := tests.Kubectl("get deployment -o=jsonpath='{.spec.replicas}' -n", depNS, depName)
+			assert.NoError(t, err)
+			assert.Equal(t, "'2'", string(depReplicas))
+
+			// Patch deployment replicas to 3 using patch file in preparation for ignore changes to be tested in step2.
+			_, err = tests.Kubectl("patch --field-manager replica/manager deployment -n", depNS, depName, "--patch-file", filepath.Join("ignore-changes", "deployment-patch.yaml"))
+			assert.NoError(t, err)
+			depReplicas, err = tests.Kubectl("get deployment -o=jsonpath='{.spec.replicas}' -n", depNS, depName)
+			assert.NoError(t, err)
+			assert.Equal(t, "'3'", string(depReplicas))
+		},
+		EditDirs: []integration.EditDir{
+			{
+				// Repeat step1 again where no changes are made to the deployment since we ignore changes to spec.replicas.
+				Dir:      filepath.Join("ignore-changes", "step1"),
+				Additive: true,
+				ExtraRuntimeValidation: func(t *testing.T, stackInfo integration.RuntimeValidationStackInfo) {
+					// Validate replicas was not updated back to 1.
+					depReplicas, err := tests.Kubectl("get deployment -o=jsonpath='{.spec.replicas}' -n", depNS, depName)
+					assert.NoError(t, err)
+					assert.Equal(t, "'3'", string(depReplicas))
+				},
+			},
+			{
+				Dir:      filepath.Join("ignore-changes", "step2"),
+				Additive: true,
+				ExtraRuntimeValidation: func(t *testing.T, stackInfo integration.RuntimeValidationStackInfo) {
+					// Validate image was updated, but spec.replicas was not.
+					depImage, err := tests.Kubectl("get deployment -o=jsonpath='{.spec.template.spec.containers[0].image}' -n", depNS, depName)
+					assert.NoError(t, err)
+					assert.Equal(t, "'nginx:1.25.1'", string(depImage))
+
+					depReplicas, err := tests.Kubectl("get deployment -o=jsonpath='{.spec.replicas}' -n", depNS, depName)
+					assert.NoError(t, err)
+					assert.Equal(t, "'3'", string(depReplicas))
+
+					// Now use kubectl patch to update spec.replicas to 5 and see if we can correctly ignore changes to spec.replicas again when the field manager is
+					// "kubectl-patch" since we have logic to override certain field managers with manager name prefixes.
```

Comment:
Yes. `fluxssa.PatchReplaceFieldManagers` in https://github.com/pulumi/pulumi-kubernetes/blob/d2a3b8c217e73c874e294f9f7cbdbef9dfe5e204/provider/pkg/await/await.go#L498C39-L532 matches field manager names with a name prefix rather than an exact match. So Pulumi will take ownership of `kubectl-patch` managed fields, not just fields managed by `kubectl`.
I've expanded the comment to make this clearer.